File: app/src/main/java/com/mahesh/mvpsample/detail/DetailModelInteractor.java
package com.mahesh.mvpsample.detail;

import com.mahesh.mvpsample.detail.adapter.DetailAdapterModel;

import java.util.ArrayList;
import java.util.List;

public class DetailModelInteractor implements DetailContract.ModelInteractor {

    private List<DetailAdapterModel> adapterModelList;

    DetailModelInteractor() {
        this.adapterModelList = new ArrayList<>();
    }

    @Override
    public void createAdapterEntityList(DetailContract.CreateDataCallback callback) {
        adapterModelList.clear();
        for (int i = 0; i < 50; i++) {
            adapterModelList.add(new DetailAdapterModel("Dummy Text " + i));
        }
        callback.onDataCreated();
    }

    @Override
    public List<DetailAdapterModel> getAdapterEntityList() {
        return adapterModelList;
    }
}

File: app/src/main/java/com/mahesh/mvpsample/detail/DetailModelInteractor.kt
package com.mahesh.mvpsample.detail

import com.mahesh.mvpsample.detail.adapter.DetailAdapterModel

class DetailModelInteractor internal constructor() : DetailContract.ModelInteractor {
    private val adapterModelList: MutableList<DetailAdapterModel>

    init {
        adapterModelList = ArrayList()
    }

    override fun createAdapterEntityList(callback: () -> Unit) {
        adapterModelList.clear()
        for (i in 0..49) {
            adapterModelList.add(DetailAdapterModel("Dummy Text $i"))
        }
        callback.invoke()
    }

    override val adapterEntityList: List<DetailAdapterModel>
        get() = adapterModelList
}
